### PR TITLE
Add release notes and update license dependencies for 6.2.1

### PR DESCRIPTION
**Pull request by rmstar**

### CODE REVIEW

File: cdap-docs/reference-manual/source/release-notes.rst [hunk 1]
```diff
@@ -30,6 +30,141 @@ Cask Data Application Platform Release Notes
...
+
+
+New Features
+------------
+
+- :cask-issue:`CDAP-16708` - Added a new AutoJoiner API for plugins to implement. The new API leaves implementation details up to the application, which can perform join optimizations that were not possible with the older Joiner API.
+
+- :cask-issue:`CDAP-16530` - Fixed joiner output schema generation to be deterministic, using the same ordering as they had in the input data.
+
+- :cask-issue:`CDAP-16711` - Added the ability for Joiner plugins to specify whether null keys should match other null keys.
```
Comment (bdmogal) [inline cdap-docs/reference-manual/source/release-notes.rst]: I think this is an improvement.

File: cdap-docs/reference-manual/source/release-notes.rst [hunk 2]
```diff
@@ -30,6 +30,141 @@ Cask Data Application Platform Release Notes
...
+
+- :cask-issue:`CDAP-16875` - Added two new properties to the Joiner plugin -- 'Join on Null Values' and 'Inputs to Load in Memory'. The plugin also uses new join capabilities that make it more resilient to running out of memory.
```
Comment (bdmogal) [inline cdap-docs/reference-manual/source/release-notes.rst]: this is already covered in the previous two right?

File: cdap-docs/reference-manual/source/release-notes.rst [hunk 3]
```diff
@@ -30,6 +30,141 @@ Cask Data Application Platform Release Notes
...
+
+- :cask-issue:`CDAP-16461` - Added Spark parameter to limit Spark block size to prevent issues with joins
+
+- :cask-issue:`CDAP-16855` - Introduced a new aggregator API to achieve better performance for spark engine.
```
Comment (bdmogal) [inline cdap-docs/reference-manual/source/release-notes.rst]: ... performance when using Apache Spark engine

File: cdap-docs/reference-manual/source/release-notes.rst [hunk 4]
```diff
@@ -30,6 +30,141 @@ Cask Data Application Platform Release Notes
...
+
+- :cask-issue:`CDAP-16708` - Added a new AutoJoiner API for plugins to implement. The new API leaves implementation details up to the application, which can perform join optimizations that were not possible with the older Joiner API.
+
+- :cask-issue:`CDAP-16530` - Fixed joiner output schema generation to be deterministic, using the same ordering as they had in the input data.
+
+- :cask-issue:`CDAP-16711` - Added the ability for Joiner plugins to specify whether null keys should match other null keys.
+
+- :cask-issue:`CDAP-16875` - Added two new properties to the Joiner plugin -- 'Join on Null Values' and 'Inputs to Load in Memory'. The plugin also uses new join capabilities that make it more resilient to running out of memory.
+
+- :cask-issue:`CDAP-16461` - Added Spark parameter to limit Spark block size to prevent issues with joins
```
Comment (bdmogal) [inline cdap-docs/reference-manual/source/release-notes.rst]: improvement?

File: cdap-docs/reference-manual/source/release-notes.rst [hunk 5]
```diff
@@ -30,6 +30,141 @@ Cask Data Application Platform Release Notes
...
+
+- :cask-issue:`CDAP-16855` - Introduced a new aggregator API to achieve better performance for spark engine.
+
+
+Improvements
+------------
+
+- :cask-issue:`CDAP-16455` - Include logs emitted from the job main class as the Dataproc job logs
+
+- :cask-issue:`CDAP-16682` - When backend is slow to respond to requests from UI, we now show a snackbar saying there's a delay.
```
Comment (bdmogal) [inline cdap-docs/reference-manual/source/release-notes.rst]: ... the UI now shows a delay notification.

File: cdap-docs/reference-manual/source/release-notes.rst [hunk 6]
```diff
@@ -30,6 +30,141 @@ Cask Data Application Platform Release Notes
...
+
+- :cask-issue:`CDAP-16690` - Added revamped preview tab with new Record view for large schemas.
```
Comment (bdmogal) [inline cdap-docs/reference-manual/source/release-notes.rst]: Revamped the preview tab...

File: cdap-docs/reference-manual/source/release-notes.rst [hunk 7]
```diff
@@ -30,6 +30,141 @@ Cask Data Application Platform Release Notes
...
+
+- :cask-issue:`CDAP-16835` - Added support for upgrading applications via REST API. Example usage is to upgrade all pipelines in a namespace to use latest available artifacts.
+
+- :cask-issue:`CDAP-16836` - -l option is introduced in CDAP CLI to take in URI in the new format "http[s]://hostname:[port]
```
Comment (bdmogal) [inline cdap-docs/reference-manual/source/release-notes.rst]: Introduced a ``-l`` option in the CLI...

File: cdap-docs/reference-manual/source/release-notes.rst [hunk 8]
```diff
@@ -30,6 +30,141 @@ Cask Data Application Platform Release Notes
...
+
+- :cask-issue:`CDAP-16980` - New Log Viewer feature which enable user to see the most recent logs
```
Comment (bdmogal) [inline cdap-docs/reference-manual/source/release-notes.rst]: Log viewer now allows users to see the most recent logs

File: cdap-docs/reference-manual/source/release-notes.rst [hunk 9]
```diff
@@ -30,6 +30,141 @@ Cask Data Application Platform Release Notes
...
+
+- :cask-issue:`CDAP-16328` - Labeling Dataproc clusters configured as Remote Hadoop Provisioners
```
Comment (bdmogal) [inline cdap-docs/reference-manual/source/release-notes.rst]: this change was not OSS right, we should remove it.

File: cdap-docs/reference-manual/source/release-notes.rst [hunk 10]
```diff
@@ -30,6 +30,141 @@ Cask Data Application Platform Release Notes
...
+
+- :cask-issue:`CDAP-16606` - Limit to reading in 100 records across all input partitions in preview
+
+- :cask-issue:`CDAP-16621` - Removed modal showing pipeline JSON when users export pipelines. Instead, pipeline gets downloaded when users click "export pipeline" without the extra confirmation step. 
+
+- :cask-issue:`CDAP-16673` - Added payload compression support to messaging service
+
+- :cask-issue:`CDAP-16676` - Upgrade to use dataproc API v1beta2 to allow endpoint config
```
Comment (bdmogal) [inline cdap-docs/reference-manual/source/release-notes.rst]: Dataproc

File: cdap-docs/reference-manual/source/release-notes.rst [hunk 11]
```diff
@@ -30,6 +30,141 @@ Cask Data Application Platform Release Notes
...
+
+- :cask-issue:`CDAP-16697` - Created Best Practices guide for Spark engine tuning
```
Comment (bdmogal) [inline cdap-docs/reference-manual/source/release-notes.rst]: remove. this is not shipped with the code right.

File: cdap-docs/reference-manual/source/release-notes.rst [hunk 12]
```diff
@@ -30,6 +30,141 @@ Cask Data Application Platform Release Notes
...
+
+- :cask-issue:`CDAP-16709` - Implemented performance improvements to joiner plugins to cap the required memory to around 4gb per executor instead of scaling up as the skewness of the join goes up. Joins can now also be performed in-memory if one side is small, and behavior on null keys can be chosen by the user.
```
Comment (bdmogal) [inline cdap-docs/reference-manual/source/release-notes.rst]: nit: can you break the line at 120 characters?

File: cdap-docs/reference-manual/source/release-notes.rst [hunk 13]
```diff
@@ -30,6 +30,141 @@ Cask Data Application Platform Release Notes
...
+
+- :cask-issue:`CDAP-16815` - Added a metric records.updated in BigQuery sink. This will give a total of all the inserts, updates and upserts into the sink.
```
Comment (bdmogal) [inline cdap-docs/reference-manual/source/release-notes.rst]: ... This counts the total of ....

File: cdap-docs/reference-manual/source/release-notes.rst [hunk 14]
```diff
@@ -30,6 +30,141 @@ Cask Data Application Platform Release Notes
...
+
+- :cask-issue:`CDAP-16918` - Introduced a new REST API for getting all application details across all namespaces
```
Comment (bdmogal) [inline cdap-docs/reference-manual/source/release-notes.rst]: new feature

File: cdap-docs/reference-manual/source/release-notes.rst [hunk 15]
```diff
@@ -30,6 +30,141 @@ Cask Data Application Platform Release Notes
...
+
+- :cask-issue:`CDAP-16929` - Added the ability to select a Custom Dataproc Image. The complete URI for the custom image should be specified.
+
+- :cask-issue:`CDAP-16975` - UI now adds the latest version of plugin, among the list of different versions of the plugin, when added from the sidepanel in pipeline studio. If the user has already chosen a specific version (older version) it defaults to that instead of the latest.
```
Comment (bdmogal) [inline cdap-docs/reference-manual/source/release-notes.rst]: break at 120 characters

File: cdap-docs/reference-manual/source/release-notes.rst [hunk 16]
```diff
@@ -30,6 +30,141 @@ Cask Data Application Platform Release Notes
...
+
+- :cask-issue:`CDAP-16976` - UI resets the default version of plugins for specific user during upgrade. When users upgrade from 6.1.2 to 6.1.3 or later UI will reset the default version of plugin the user has already chosen. Post upgrade if the user uses the same plugin UI will choose the latest version of the same plugin.
```
Comment (bdmogal) [inline cdap-docs/reference-manual/source/release-notes.rst]: ditto

File: cdap-docs/reference-manual/source/release-notes.rst [hunk 17]
```diff
@@ -30,6 +30,141 @@ Cask Data Application Platform Release Notes
...
+
+
+Bug Fixes
+---------
+
+- :cask-issue:`CDAP-17003` - Removed redundant validations from BQ sink , this should reduce calls to BigQuery.getTable()
+
+- :cask-issue:`CDAP-16530` - Fixed joiner output schema generation to be deterministic, using the same ordering as they had in the input data.
+
+- :cask-issue:`CDAP-17016` - Verified on the change in the delta replicator. Whenever a pipeline was created and ran, the target table created in BQ side will take the primary key as clustering columns as expected.
```
Comment (bdmogal) [inline cdap-docs/reference-manual/source/release-notes.rst]: remove

File: cdap-docs/reference-manual/source/release-notes.rst [hunk 18]
```diff
@@ -30,6 +30,141 @@ Cask Data Application Platform Release Notes
...
+Bug Fixes
+---------
+
+- :cask-issue:`CDAP-17003` - Removed redundant validations from BQ sink , this should reduce calls to BigQuery.getTable()
+
+- :cask-issue:`CDAP-16530` - Fixed joiner output schema generation to be deterministic, using the same ordering as they had in the input data.
+
+- :cask-issue:`CDAP-17016` - Verified on the change in the delta replicator. Whenever a pipeline was created and ran, the target table created in BQ side will take the primary key as clustering columns as expected. 
+
+- :cask-issue:`CDAP-15869` - Preview displays logical types as strings.
```
Comment (bdmogal) [inline cdap-docs/reference-manual/source/release-notes.rst]: Fixed preview to display logical types as strings

File: cdap-docs/reference-manual/source/release-notes.rst [hunk 19]
```diff
@@ -30,6 +30,141 @@ Cask Data Application Platform Release Notes
...
+
+- :cask-issue:`CDAP-16222` - Fixed the package references in the dynamic spark plugin to use io.cdap instead of co.cask.
+
+- :cask-issue:`CDAP-16340` - Fixed the joiner plugin to allow a nullable key on one side and a non-nullable key on the other.
+
+- :cask-issue:`CDAP-16367` - Fixed a bug where field lineage is incorrect when a source is directly connected to a sink.
+
+- :cask-issue:`CDAP-16487` - Fixed regex for empty filter in Wrangler UI
+
+- :cask-issue:`CDAP-16731` - Fixed a bug that the groupBy aggregator requires a different alias for the field name.
```
Comment (bdmogal) [inline cdap-docs/reference-manual/source/release-notes.rst]: GroupBy

File: cdap-docs/reference-manual/source/release-notes.rst [hunk 20]
```diff
@@ -30,6 +30,141 @@ Cask Data Application Platform Release Notes
...
+
+- :cask-issue:`CDAP-16747` - Modified loading screen for preview tab.
```
Comment (bdmogal) [inline cdap-docs/reference-manual/source/release-notes.rst]: if there is no more information about this, let's remove this

File: cdap-docs/reference-manual/source/release-notes.rst [hunk 21]
```diff
@@ -30,6 +30,141 @@ Cask Data Application Platform Release Notes
...
+
+- :cask-issue:`CDAP-16487` - Fixed regex for empty filter in Wrangler UI
+
+- :cask-issue:`CDAP-16731` - Fixed a bug that the groupBy aggregator requires a different alias for the field name.
+
+- :cask-issue:`CDAP-16747` - Modified loading screen for preview tab. 
+
+- :cask-issue:`CDAP-16760` - Fixed a bug where memory, cpu, and engine config properties were not being set for sparkprogram plugins
+
+- :cask-issue:`CDAP-16786` - Fixes listing pipelines by tags in Pipelines list UI.
```
Comment (bdmogal) [inline cdap-docs/reference-manual/source/release-notes.rst]: nit: s/UI/page/

File: cdap-docs/reference-manual/source/release-notes.rst [hunk 22]
```diff
@@ -30,6 +30,141 @@ Cask Data Application Platform Release Notes
...
+
+- :cask-issue:`CDAP-16870` - Fixed PySpark support to work with Spark 2.1.3+
+
+- :cask-issue:`CDAP-16879` - 'Truncate table' and 'update schema' options if set together, will apply only WRITE_TRUNCATE to BQ job.
+
+- :cask-issue:`CDAP-16880` - Removed schema validation from BQ sink when 'truncate table' option is set. 
+
+- :cask-issue:`CDAP-16891` - Unsupported pipelines in drafts would be upgraded when users open them.
+
+- :cask-issue:`CDAP-16927` - Verified on the latest version of Azure Blob Store Plugin. If the account name is not ended with ".blob.core.windows.net", the validation will fail with error message to tell user to add that suffix.
```
Comment (bdmogal) [inline cdap-docs/reference-manual/source/release-notes.rst]: Added validation to ensure that account name ends with ".blob.core.windows.net" in the Azure Blob Store plugin

File: cdap-docs/reference-manual/source/release-notes.rst [hunk 23]
```diff
@@ -30,6 +30,141 @@ Cask Data Application Platform Release Notes
...
+
+- :cask-issue:`CDAP-16972` - Fixed an issue where preview config would open when trying to stop a preview.
+
+- :cask-issue:`CDAP-16993` - Fixed a bug in preview for fields that have non-string types such as bytes. 
+
+- :cask-issue:`CDAP-17029` - Fixed an issue that caused an extra empty row to appear when sampling GCS text files in Wrangler
+
+- :cask-issue:`CDAP-17043` - Fixes the bug for showing dropdown menu for wrangler tabs to be correct. Existing dropdown overlapped with other UI elements hindering the usage of UI.
+
+- :cask-issue:`CDAP-17045` - Fixes the bug to allow large pipelines with `-` in the name to properly overflow in UI.
```
Comment (bdmogal) [inline cdap-docs/reference-manual/source/release-notes.rst]: properly overflow seems wrong? should it be properly render?

File: cdap-docs/reference-manual/source/release-notes.rst [hunk 24]
```diff
@@ -30,6 +30,141 @@ Cask Data Application Platform Release Notes
...
+
+- :cask-issue:`CDAP-17074` - Improved state transitions for starting pipelines in app fabric to increase stability if app fabric unexpectedly restarts
```
Comment (bdmogal) [inline cdap-docs/reference-manual/source/release-notes.rst]: in the App Fabric system service...

**Review by bdmogal — CHANGES_REQUESTED** [review]

File: cdap-docs/reference-manual/source/release-notes.rst (hunk 1 above)
Comment (rmstar) [reply cdap-docs/reference-manual/source/release-notes.rst]: fixed.

File: cdap-docs/reference-manual/source/release-notes.rst (hunk 2 above)
Comment (rmstar) [reply cdap-docs/reference-manual/source/release-notes.rst]: removed.

File: cdap-docs/reference-manual/source/release-notes.rst (hunk 3 above)
Comment (rmstar) [reply cdap-docs/reference-manual/source/release-notes.rst]: fixed

File: cdap-docs/reference-manual/source/release-notes.rst (hunk 4 above)
Comment (rmstar) [reply cdap-docs/reference-manual/source/release-notes.rst]: fixed

File: cdap-docs/reference-manual/source/release-notes.rst (hunk 5 above)
Comment (rmstar) [reply cdap-docs/reference-manual/source/release-notes.rst]: fixed

File: cdap-docs/reference-manual/source/release-notes.rst (hunk 6 above)
Comment (rmstar) [reply cdap-docs/reference-manual/source/release-notes.rst]: fixed.

File: cdap-docs/reference-manual/source/release-notes.rst (hunk 7 above)
Comment (rmstar) [reply cdap-docs/reference-manual/source/release-notes.rst]: fixed.

File: cdap-docs/reference-manual/source/release-notes.rst (hunk 8 above)
Comment (rmstar) [reply cdap-docs/reference-manual/source/release-notes.rst]: fixed

File: cdap-docs/reference-manual/source/release-notes.rst (hunk 9 above)
Comment (rmstar) [reply cdap-docs/reference-manual/source/release-notes.rst]: removed.

File: cdap-docs/reference-manual/source/release-notes.rst (hunk 10 above)
Comment (rmstar) [reply cdap-docs/reference-manual/source/release-notes.rst]: fixed

File: cdap-docs/reference-manual/source/release-notes.rst (hunk 11 above)
Comment (rmstar) [reply cdap-docs/reference-manual/source/release-notes.rst]: removed.

File: cdap-docs/reference-manual/source/release-notes.rst (hunk 12 above)
Comment (rmstar) [reply cdap-docs/reference-manual/source/release-notes.rst]: fixed

File: cdap-docs/reference-manual/source/release-notes.rst (hunk 13 above)
Comment (rmstar) [reply cdap-docs/reference-manual/source/release-notes.rst]: fixed

File: cdap-docs/reference-manual/source/release-notes.rst (hunk 15 above)
Comment (rmstar) [reply cdap-docs/reference-manual/source/release-notes.rst]: fixed

File: cdap-docs/reference-manual/source/release-notes.rst (hunk 16 above)
Comment (rmstar) [reply cdap-docs/reference-manual/source/release-notes.rst]: fixed

File: cdap-docs/reference-manual/source/release-notes.rst (hunk 17 above)
Comment (rmstar) [reply cdap-docs/reference-manual/source/release-notes.rst]: removed

File: cdap-docs/reference-manual/source/release-notes.rst (hunk 18 above)
Comment (rmstar) [reply cdap-docs/reference-manual/source/release-notes.rst]: fixed

File: cdap-docs/reference-manual/source/release-notes.rst (hunk 19 above)
Comment (rmstar) [reply cdap-docs/reference-manual/source/release-notes.rst]: fixed

File: cdap-docs/reference-manual/source/release-notes.rst (hunk 20 above)
Comment (rmstar) [reply cdap-docs/reference-manual/source/release-notes.rst]: removed

File: cdap-docs/reference-manual/source/release-notes.rst (hunk 21 above)
Comment (rmstar) [reply cdap-docs/reference-manual/source/release-notes.rst]: fixed

File: cdap-docs/reference-manual/source/release-notes.rst (hunk 22 above)
Comment (rmstar) [reply cdap-docs/reference-manual/source/release-notes.rst]: fixed

File: cdap-docs/reference-manual/source/release-notes.rst (hunk 24 above)
Comment (rmstar) [reply cdap-docs/reference-manual/source/release-notes.rst]: fixed

**Review by bdmogal — APPROVED** [review]
The release notes look good.

**Pull request merged**